### PR TITLE
SCML: Raise ValueError if n_features larger than n_triplets

### PR DESCRIPTION
**Pull request by nikosmichas**
Related discussion https://github.com/scikit-learn-contrib/metric-learn/issues/348
cc @grudloff

### CODE REVIEW

File: metric_learn/scml.py [hunk 1]
```diff
@@ -240,6 +240,12 @@ def _generate_bases_dist_diff(self, triplets, X):
       raise ValueError("n_basis should be an integer, instead it is of type %s"
                        % type(self.n_basis))
 
+    if n_features > n_triplets:
+      raise ValueError(
+        "Number of features (%s) is greater than the nuber of triplets(%s).\n"
```
Comment (grudloff) [inline metric_learn/scml.py:245]: small typo

Comment (nikosmichas) [reply metric_learn/scml.py]: fixed

File: metric_learn/scml.py [hunk 2]
```diff
@@ -240,6 +240,12 @@ def _generate_bases_dist_diff(self, triplets, X):
...
+        "Consider using a dimensionality reduction preprocessing or create "
+        "a new basis generation scheme." % (n_features, n_triplets))
+
```
[the remark this answers is not in the document]
Comment (nikosmichas) [reply metric_learn/scml.py:248]: thanks, updated

**Review by bellet — CHANGES_REQUESTED** [review]
Thanks for the work @nikosmichas !
This looks good to me, except that I realize that the docstring entry for the `‘triplet_diffs’` option says:

```
The basis set is constructed from the differences between points of n_basis positive or negative pairs taken from the triplets constrains.
```

Shoudn't it be something like:

```
Each basis is constructed from the differences between points of n_features positive or negative pairs taken randomly from the triplets constraints.
```

**Comment by nikosmichas** [comment]
> Thanks for the work @nikosmichas ! This looks good to me, except that I realize that the docstring entry for the `‘triplet_diffs’` option says:
> 
> ```
> The basis set is constructed from the differences between points of n_basis positive or negative pairs taken from the triplets constrains.
> ```
> 
> Shoudn't it be something like:
> 
> ```
> Each basis is constructed from the differences between points of n_features positive or negative pairs taken randomly from the triplets constraints.
> ```

@bellet not sure about the exact wording. I can make the docstring change if you think that it's better. Do you thing that the change is related to this PR ?

**Comment by perimosocordiae** [comment]
Merging now, we can fix the docstring separately.

**Comment by bellet** [comment]
Thanks @nikosmichas !

**Pull request merged**